### ISSUE TITLE
unrecognized arguments: --cluster-autoscaler-profile scan-interval=30s

### ISSUE THREAD

**Issue by AdnanHodzic** [Azure/azure-cli#13030]
## Describe the bug

**Command Name**
```
az aks create
```

**Errors:**
```
az: error: unrecognized arguments: --cluster-autoscaler-profile scan-interval=30s
```

## To Reproduce:

```
az aks create \
  --resource-group xyz \
  --name xyz \
  --node-vm-size Standard_F2s_v2 \
  --nodepool-name xyz \
  --node-count 1 \
  --enable-cluster-autoscaler \
  --min-count 1\
  --max-count 3 \
  --cluster-autoscaler-profile scan-interval=30s \
  --location westeurope \
  --enable-addons monitoring \
  --generate-ssh-keys \
  --service-principal xyz \
  --client-secret xyz
```
Will result in above mentioned error. 

If I simply omit `--cluster-autoscaler-profile scan-interval=30s` cluster will be successfully created.

This option is suggested in documentation page: [Set the cluster autoscaler profile when creating an AKS](https://docs.microsoft.com/en-us/azure/aks/cluster-autoscaler#set-the-cluster-autoscaler-profile-when-creating-an-aks-cluster)

```
az aks create \
  --resource-group myResourceGroup \
  --name myAKSCluster \
  --node-count 1 \
  --enable-cluster-autoscaler \
  --min-count 1 \
  --max-count 3 \
  --cluster-autoscaler-profile scan-interval=30s
```

Furthermore, if I try to add `scan-itnerval` option once the cluster was created as suggested in documentation: [Set the cluster autoscaler profile on an existing AKS cluster](https://docs.microsoft.com/en-us/azure/aks/cluster-autoscaler#set-the-cluster-autoscaler-profile-on-an-existing-aks-cluster)

```
az aks update \
  --resource-group xyz \
  --name xyz \
  --cluster-autoscaler-profile scan-interval=30s
```

I'll get the same error:

```
az: error: unrecognized arguments: --cluster-autoscaler-profile scan-interval=30s
```

## Expected Behavior

I would expect `scan-interval` option which defines how often cluster is reevaluated for scale up or down to be enabled as [suggested in official documentation.](https://docs.microsoft.com/en-us/azure/aks/cluster-autoscaler)

## Environment Summary
```
Linux-5.4.0-21-generic-x86_64-with-debian-bullseye-sid (Ubuntu 20.04)
Python 3.6.5
Installer: DEB

azure-cli 2.3.1

```

**Comment by yungezz** [comment]
routing to appropriate team

**Comment by marwanad** [comment]
@AdnanHodzic are you using CLI extensions? This feature (setting the autoscaler profile) is only available in the preview CLI extension:

```
az extension add --name aks-preview
```

**Comment by AdnanHodzic** [comment]
@marwanad I wasn't. But now that I tried I have a problem with installing `aks-preview` extension, namely:

```
~$ az extension add --name aks-preview --debug
Command arguments: ['extension', 'add', '--name', 'aks-preview', '--debug']
Event: Cli.PreExecute []
Event: CommandParser.OnGlobalArgumentsCreate [<function CLILogging.on_global_arguments at 0x7f9a32e74a60>, <function OutputProducer.on_global_arguments at 0x7f9a32dc5158>, <function CLIQuery.on_global_arguments at 0x7f9a32d64158>]
Event: CommandInvoker.OnPreCommandTableCreate []
Installed command modules ['acr', 'acs', 'advisor', 'ams', 'apim', 'appconfig', 'appservice', 'backup', 'batch', 'batchai', 'billing', 'botservice', 'cdn', 'cloud', 'cognitiveservices', 'configure', 'consumption', 'container', 'cosmosdb', 'deploymentmanager', 'dla', 'dls', 'dms', 'eventgrid', 'eventhubs', 'extension', 'feedback', 'find', 'hdinsight', 'interactive', 'iot', 'iotcentral', 'keyvault', 'kusto', 'lab', 'managedservices', 'maps', 'monitor', 'natgateway', 'netappfiles', 'network', 'policyinsights', 'privatedns', 'profile', 'rdbms', 'redis', 'relay', 'reservations', 'resource', 'role', 'search', 'security', 'servicebus', 'servicefabric', 'signalr', 'sql', 'sqlvm', 'storage', 'vm']
Loaded module 'acr' in 0.009 seconds.
Loaded module 'acs' in 0.023 seconds.
Loaded module 'advisor' in 0.001 seconds.
Event: CommandLoader.OnLoadCommandTable []
Loaded module 'ams' in 0.008 seconds.
Loaded module 'apim' in 0.002 seconds.
Loaded module 'appconfig' in 0.003 seconds.
Loaded module 'appservice' in 0.014 seconds.
Loaded module 'backup' in 0.005 seconds.
Event: CommandLoader.OnLoadCommandTable []
Loaded module 'batch' in 0.011 seconds.
Loaded module 'batchai' in 0.004 seconds.
Loaded module 'billing' in 0.002 seconds.
Loaded module 'botservice' in 0.005 seconds.
Event: CommandLoader.OnLoadCommandTable []
Loaded module 'cdn' in 0.006 seconds.
Loaded module 'cloud' in 0.003 seconds.
Loaded module 'cognitiveservices' in 0.002 seconds.
Loaded module 'configure' in 0.001 seconds.
Loaded module 'consumption' in 0.003 seconds.
Loaded module 'container' in 0.002 seconds.
Loaded module 'cosmosdb' in 0.009 seconds.
Loaded module 'deploymentmanager' in 0.003 seconds.
Loaded module 'dla' in 0.006 seconds.
Loaded module 'dls' in 0.004 seconds.
Loaded module 'dms' in 0.003 seconds.
Loaded module 'eventgrid' in 0.003 seconds.
Loaded module 'eventhubs' in 0.006 seconds.
Loaded module 'extension' in 0.001 seconds.
Loaded module 'feedback' in 0.001 seconds.
Loaded module 'find' in 0.001 seconds.
Loaded module 'hdinsight' in 0.003 seconds.
Loaded module 'interactive' in 0.000 seconds.
Loaded module 'iot' in 0.006 seconds.
Loaded module 'iotcentral' in 0.002 seconds.
Loaded module 'keyvault' in 0.008 seconds.
Loaded module 'kusto' in 0.002 seconds.
Loaded module 'lab' in 0.004 seconds.
Loaded module 'managedservices' in 0.003 seconds.
Loaded module 'maps' in 0.002 seconds.
Loaded module 'monitor' in 0.010 seconds.
Loaded module 'natgateway' in 0.002 seconds.
Event: CommandLoader.OnLoadCommandTable []
Loaded module 'netappfiles' in 0.004 seconds.
Loaded module 'network' in 0.048 seconds.
Loaded module 'policyinsights' in 0.002 seconds.
Loaded module 'privatedns' in 0.007 seconds.
Loaded module 'profile' in 0.003 seconds.
Loaded module 'rdbms' in 0.010 seconds.
Loaded module 'redis' in 0.003 seconds.
Loaded module 'relay' in 0.004 seconds.
Loaded module 'reservations' in 0.003 seconds.
Loaded module 'resource' in 0.015 seconds.
Loaded module 'role' in 0.007 seconds.
Loaded module 'search' in 0.002 seconds.
Loaded module 'security' in 0.003 seconds.
Loaded module 'servicebus' in 0.007 seconds.
Loaded module 'servicefabric' in 0.005 seconds.
Loaded module 'signalr' in 0.002 seconds.
Loaded module 'sql' in 0.011 seconds.
Loaded module 'sqlvm' in 0.003 seconds.
Event: CommandLoader.OnLoadCommandTable []
Loaded module 'storage' in 0.047 seconds.
Loaded module 'vm' in 0.022 seconds.
Loaded all modules in 0.388 seconds. (note: there's always an overhead with the first module loaded)
Extensions directory: '/home/xyz/.azure/cliextensions'
Event: CommandInvoker.OnPreCommandTableTruncate [<function AzCliLogging.init_command_file_logging at 0x7f9a32aec7b8>]
az_command_data_logger : command args: extension add --name {} --debug
metadata file logging enabled - writing logs to '/home/xyz/.azure/commands'.
Event: CommandInvoker.OnPreArgumentLoad [<function register_global_subscription_argument.<locals>.add_subscription_parameter at 0x7f9a32b0a510>]
Event: CommandInvoker.OnPostArgumentLoad []
Event: CommandInvoker.OnPostCommandTableCreate [<function register_ids_argument.<locals>.add_ids_arguments at 0x7f9a32ac4378>, <function register_cache_arguments.<locals>.add_cache_arguments at 0x7f9a32ac4510>]
Event: CommandInvoker.OnCommandTableLoaded []
Event: CommandInvoker.OnPreParseArgs [<function _documentdb_deprecate at 0x7f9a312b3e18>]
Event: CommandInvoker.OnPostParseArgs [<function OutputProducer.handle_output_argument at 0x7f9a32dc51e0>, <function CLIQuery.handle_query_parameter at 0x7f9a32d641e0>, <function register_ids_argument.<locals>.parse_ids_arguments at 0x7f9a32ac4400>, <function handler at 0x7f9a3110f9d8>]
Extensions directory: '/home/xyz/.azure/cliextensions'
urllib3.connectionpool : Starting new HTTPS connection (1): aka.ms:443
urllib3.connectionpool : https://aka.ms:443 "GET /azure-cli-extension-index-v1 HTTP/1.1" 301 0
urllib3.connectionpool : Starting new HTTPS connection (1): azurecliextensionsync.blob.core.windows.net:443
urllib3.connectionpool : https://azurecliextensionsync.blob.core.windows.net:443 "GET /index1/index.json HTTP/1.1" 200 231992
Candidates ['aks_preview-0.4.35-py2.py3-none-any.whl', 'aks_preview-0.4.36-py2.py3-none-any.whl', 'aks_preview-0.4.37-py2.py3-none-any.whl', 'aks_preview-0.4.38-py2.py3-none-any.whl', 'aks_preview-0.4.39-py2.py3-none-any.whl', 'aks_preview-0.4.40-py2.py3-none-any.whl', 'aks_preview-0.4.41-py2.py3-none-any.whl', 'aks_preview-0.4.42-py2.py3-none-any.whl', 'aks_preview-0.4.43-py2.py3-none-any.whl']
Candidates ['aks_preview-0.4.35-py2.py3-none-any.whl', 'aks_preview-0.4.36-py2.py3-none-any.whl', 'aks_preview-0.4.37-py2.py3-none-any.whl', 'aks_preview-0.4.38-py2.py3-none-any.whl', 'aks_preview-0.4.39-py2.py3-none-any.whl', 'aks_preview-0.4.40-py2.py3-none-any.whl', 'aks_preview-0.4.41-py2.py3-none-any.whl', 'aks_preview-0.4.42-py2.py3-none-any.whl', 'aks_preview-0.4.43-py2.py3-none-any.whl']
Candidates ['aks_preview-0.4.35-py2.py3-none-any.whl', 'aks_preview-0.4.36-py2.py3-none-any.whl', 'aks_preview-0.4.37-py2.py3-none-any.whl', 'aks_preview-0.4.38-py2.py3-none-any.whl', 'aks_preview-0.4.39-py2.py3-none-any.whl', 'aks_preview-0.4.40-py2.py3-none-any.whl', 'aks_preview-0.4.41-py2.py3-none-any.whl', 'aks_preview-0.4.42-py2.py3-none-any.whl', 'aks_preview-0.4.43-py2.py3-none-any.whl']
Candidates ['aks_preview-0.4.43-py2.py3-none-any.whl', 'aks_preview-0.4.42-py2.py3-none-any.whl', 'aks_preview-0.4.41-py2.py3-none-any.whl', 'aks_preview-0.4.40-py2.py3-none-any.whl', 'aks_preview-0.4.39-py2.py3-none-any.whl', 'aks_preview-0.4.38-py2.py3-none-any.whl', 'aks_preview-0.4.37-py2.py3-none-any.whl', 'aks_preview-0.4.36-py2.py3-none-any.whl', 'aks_preview-0.4.35-py2.py3-none-any.whl']
Choosing the latest of the remaining candidates.
Chosen {'downloadUrl': 'https://azurecliprod.blob.core.windows.net/cli-extensions/aks_preview-0.4.43-py2.py3-none-any.whl', 'filename': 'aks_preview-0.4.43-py2.py3-none-any.whl', 'metadata': {'azext.isPreview': True, 'azext.minCliCoreVersion': '2.0.49', 'classifiers': ['Development Status :: 4 - Beta', 'Intended Audience :: Developers', 'Intended Audience :: System Administrators', 'Programming Language :: Python', 'Programming Language :: Python :: 3', 'Programming Language :: Python :: 3.6', 'Programming Language :: Python :: 3.7', 'Programming Language :: Python :: 3.8', 'License :: OSI Approved :: MIT License'], 'extensions': {'python.details': {'contacts': [{'email': 'azpycli@microsoft.com', 'name': 'Microsoft Corporation', 'role': 'author'}], 'document_names': {'description': 'DESCRIPTION.rst'}, 'project_urls': {'Home': 'https://github.com/Azure/azure-cli-extensions/tree/master/src/aks-preview'}}}, 'generator': 'bdist_wheel (0.30.0)', 'license': 'MIT', 'metadata_version': '2.0', 'name': 'aks-preview', 'summary': 'Provides a preview for upcoming AKS features', 'version': '0.4.43'}, 'sha256Digest': 'd351dc69bac520400beffaa571dccdd1fd37534c28309fa703ae59802d25cad5'}
Extension source is url? True
Extensions directory: '/home/xyz/.azure/cliextensions'
Downloading https://azurecliprod.blob.core.windows.net/cli-extensions/aks_preview-0.4.43-py2.py3-none-any.whl to /tmp/tmppneur2gu/aks_preview-0.4.43-py2.py3-none-any.whl
urllib3.connectionpool : Starting new HTTPS connection (1): azurecliprod.blob.core.windows.net:443
urllib3.connectionpool : https://azurecliprod.blob.core.windows.net:443 "GET /cli-extensions/aks_preview-0.4.43-py2.py3-none-any.whl HTTP/1.1" 200 583218
Downloaded to /tmp/tmppneur2gu/aks_preview-0.4.43-py2.py3-none-any.whl
Validating the extension /tmp/tmppneur2gu/aks_preview-0.4.43-py2.py3-none-any.whl
Checksum of /tmp/tmppneur2gu/aks_preview-0.4.43-py2.py3-none-any.whl is OK
Extension compatibility result: is_compatible=True cli_core_version=2.3.1 min_required=2.0.49 max_required=None
Validation successful on /tmp/tmppneur2gu/aks_preview-0.4.43-py2.py3-none-any.whl
Linux distro check: Reading from: /etc/apt/sources.list.d/azure-cli.list
Linux distro check: Found in list file: disco
Linux distro check: Reported by API: focal
Linux distro check: Mismatch distribution name in /etc/apt/sources.list.d/azure-cli.list file
Linux distro check: If command fails, install the appropriate package for your distribution or change the above file accordingly.
Linux distro check: /etc/apt/sources.list.d/azure-cli.list has 'disco', current distro is 'focal'
Executing pip with args: ['install', '--target', '/home/xyz/.azure/cliextensions/aks-preview', '/tmp/tmppneur2gu/aks_preview-0.4.43-py2.py3-none-any.whl']
Running: ['/opt/az/bin/python3', '-m', 'pip', 'install', '--target', '/home/xyz/.azure/cliextensions/aks-preview', '/tmp/tmppneur2gu/aks_preview-0.4.43-py2.py3-none-any.whl', '-vv', '--disable-pip-version-check', '--no-cache-dir']
Traceback (most recent call last):
  File "/opt/az/lib/python3.6/runpy.py", line 183, in _run_module_as_main
    mod_name, mod_spec, code = _get_module_details(mod_name, _Error)
  File "/opt/az/lib/python3.6/runpy.py", line 142, in _get_module_details
    return _get_module_details(pkg_main_name, error)
  File "/opt/az/lib/python3.6/runpy.py", line 109, in _get_module_details
    __import__(pkg_name)
  File "/opt/az/lib/python3.6/site-packages/pip/__init__.py", line 45, in <module>
    from pip.vcs import git, mercurial, subversion, bazaar  # noqa
  File "/opt/az/lib/python3.6/site-packages/pip/vcs/mercurial.py", line 9, in <module>
    from pip.download import path_to_url
  File "/opt/az/lib/python3.6/site-packages/pip/download.py", line 36, in <module>
    from pip.utils.glibc import libc_ver
  File "/opt/az/lib/python3.6/site-packages/pip/utils/glibc.py", line 4, in <module>
    import ctypes
  File "/opt/az/lib/python3.6/ctypes/__init__.py", line 7, in <module>
    from _ctypes import Union, Structure, Array
ImportError: libffi.so.6: cannot open shared object file: No such file or directory

Command '['/opt/az/bin/python3', '-m', 'pip', 'install', '--target', '/home/xyz/.azure/cliextensions/aks-preview', '/tmp/tmppneur2gu/aks_preview-0.4.43-py2.py3-none-any.whl', '-vv', '--disable-pip-version-check', '--no-cache-dir']' returned non-zero exit status 1.
Pip failed so deleting anything we might have installed at /home/xyz/.azure/cliextensions/aks-preview
cli.azure.cli.core.util : An error occurred. Pip failed with status code 1. Use --debug for more information.
An error occurred. Pip failed with status code 1. Use --debug for more information.
az_command_data_logger : exit code: 1
telemetry.save : Save telemetry record of length 2559 in cache
telemetry.check : Negative: The /home/xyz/.azure/telemetry.txt was modified at 2020-04-17 14:23:04.724513, which in less than 600.000000 s
command ran in 16.111 seconds.
```

Any suggestions are welcome.

**Comment by marwanad** [comment]
@AdnanHodzic 
the errors point to a distro mismatch issue:
```
Linux distro check: Mismatch distribution name in /etc/apt/sources.list.d/azure-cli.list file
Linux distro check: If command fails, install the appropriate package for your distribution or change the above file accordingly.
Linux distro check: /etc/apt/sources.list.d/azure-cli.list has 'disco', current distro is 'focal'
```

As mentioned, edit `/etc/apt/sources.list.d/azure-cli.list` to use your distro, 'focal' then run `apt-get update`. Then try adding the extension again.

**Comment by AdnanHodzic** [comment]
@marwanad right, but as I mentioned in https://github.com/Azure/azure-cli/issues/9585#issuecomment-608319542 I had to use `disco` instead of `focal` since your repositories still don't support focal release.

And with `focal` Ubuntu 20.04 next LTS  release just being around the corner, I would ask you add support for it as well.

**Comment by marwanad** [comment]
@AdnanHodzic hmm, does reinstalling libffi6 work or running azure cli via docker?

The packaging issue is more of an issue with CLI.  I'd suggest you open a separate issue for that so it can be routed to the core team.

**Comment by AdnanHodzic** [comment]
@marwanad problem is that `libffi6` is not available in 20.04 `focal` repos, last Ubuntu release it was in is 19.04 `eoan`. 

However, out of sheer curiosity I [downloaded libffi6 for 19.04 release](https://packages.ubuntu.com/eoan/libffi6), installed it and it worked.

```
$ sudo dpkg -i libffi6_3.2.1-9_amd64.deb 
Selecting previously unselected package libffi6:amd64.
(Reading database ... 290781 files and directories currently installed.)
Preparing to unpack libffi6_3.2.1-9_amd64.deb ...
Unpacking libffi6:amd64 (3.2.1-9) ...
Setting up libffi6:amd64 (3.2.1-9) ...
Processing triggers for libc-bin (2.31-0ubuntu7) ...
$ az extension add --name aks-preview
The installed extension 'aks-preview' is in preview.
```

After which I was able to add `--cluster-autoscaler-profile scan-interval=30s` option to my cluster.

I created a feature request in regards to packaging issue: https://github.com/Azure/azure-cli/issues/13081

Thank you for your assistance and please feel free to close this issue.

**Comment by palma21** [comment]
Closing, and using the linked issue to track.
